src/compiler/crystal/macros.cr

```diff
@@ -1187,6 +1187,17 @@ module Crystal::Macros
   class Out < UnaryExpression
   end
 
+  # An `offsetof` expression.
+  class OffsetOf < ASTNode
+    # Returns the type that has been used in this `offsetof` expression.
+    def offsetof_type : ASTNode
```

Comment:
Just `type` perhaps? Seems a bit redundant when you have to say again that you want the type of the `offsetof` when you already call it on an `offsetof` expression.

Reply:
https://github.com/crystal-lang/crystal/pull/7589#discussion_r269084433

Reply:
@r00ster91 Good point. Though its internal name must stay `offsetof_type`: https://github.com/crystal-lang/crystal/pull/7589#discussion_r269084433